tools/UnityPaths/UnityPaths.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace UnityPaths
+{
+    public static class Paths
+    {
+        // If SpecialFolder.ProgramFiles returns the (x86) variant instead of "Program Files",
+        // check that "Prefer 32 bit" is unticked in the project's Build settings.
+        // On MacOS, SpecialFolder.ProgramFiles maps to "/Applications"
+        public static readonly string ProgramFiles = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
+        public static readonly string UnityHubPath = Path.Combine(ProgramFiles, "Unity", "Hub");
+        public static readonly string DefaultUnityPath = Path.Combine(ProgramFiles, "Unity");
+        public const string UnityHomeEnvVar = "UNITY_HOME";
+
+        public static string UnityHomeValue => Environment.GetEnvironmentVariable(UnityHomeEnvVar);
+
+        // Adjust the root based on Windows or MacOS. Linux is currently unsupported.
+        public static string ImprobableUnityRootPath =>
+            RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? "/Applications" : Path.Combine(@"c:\", "Unity");
+
+        public static string TryGetUnityPath()
+        {
+            if (!Directory.Exists("Assets"))
```

Comment:
Is there a safer way to check we are in a Unity project? This seems kinda brittle?

Reply:
We could check for "ProjectSettings" as well. Otherwise, there's not anything else we can check for.